source/ed/connectors/fusio_parser.cpp

```diff
@@ -657,6 +657,76 @@ void LineFusioHandler::handle_line(Data& data, const csv_row& row, bool is_first
     gtfs_data.line_map[line->uri] = line;
 }
 
+void LineGroupFusioHandler::init(Data&) {
+    id_c = csv.get_pos_col("line_group_id");
+    name_c = csv.get_pos_col("line_group_name");
+    main_line_id_c = csv.get_pos_col("main_line_id");
+}
+
+void LineGroupFusioHandler::handle_line(Data& data, const csv_row& row, bool) {
+    if(!(has_col(id_c, row) && has_col(name_c, row) && has_col(main_line_id_c, row))) {
+        LOG4CPLUS_WARN(logger, "LineGroupFusioHandler: Line ignored in " << csv.filename <<
+            " missing line_group_id, line_group_name or main_line_id column");
+    }
+    auto line = gtfs_data.line_map.find(row[main_line_id_c]);
+    if (line == gtfs_data.line_map.end()) {
+        LOG4CPLUS_WARN(logger, "LineGroupFusioHandler: Impossible to find the line " << row[main_line_id_c]);
+        return;
+    }
+    ed::types::LineGroup* line_group = new ed::types::LineGroup();
+    line_group->uri = row[id_c];
+    line_group->name = row[name_c];
+    line_group->main_line = line->second;
+
+    // Link main_line to line_group
+    ed::types::LineGroupLink line_group_link;
+    line_group_link.line_group = line_group;
+    line_group_link.line = line->second;
+    line_group_link.is_main_line = true;
+
+    line_group->linkedLinesURI.insert(line->second->uri);
+
+    data.line_groups.push_back(line_group);
+    data.line_group_links.push_back(line_group_link);
+    gtfs_data.line_group_map[line_group->uri] = line_group;
+}
+
+void LineGroupLinksFusioHandler::init(Data&) {
+    line_group_id_c = csv.get_pos_col("line_group_id");
+    line_id_c = csv.get_pos_col("line_id");
+}
+
+void LineGroupLinksFusioHandler::handle_line(Data& data, const csv_row& row, bool) {
+    if(!(has_col(line_group_id_c, row) && has_col(line_id_c, row))) {
+        LOG4CPLUS_WARN(logger, "LineGroupLinksFusioHandler: Line ignored in " << csv.filename <<
+        " missing line_group_id or line_id column");
+    }
+    auto line_group = gtfs_data.line_group_map.find(row[line_group_id_c]);
+    if (line_group == gtfs_data.line_group_map.end()) {
+        LOG4CPLUS_WARN(logger, "LineGroupLinksFusioHandler: Impossible to find the line_group " << row[line_group_id_c]);
+        return;
+    }
+
+    auto line = gtfs_data.line_map.find(row[line_id_c]);
+    if (line == gtfs_data.line_map.end()) {
+        LOG4CPLUS_WARN(logger, "LineGroupLinksFusioHandler: Impossible to find the line " << row[line_id_c]);
+        return;
+    }
+
+    if(line_group->second->linkedLinesURI.find(line->second->uri) != line_group->second->linkedLinesURI.end()) {
```

Comment:
not sure, but it seems that the linkedLinesURI are only used to make sure no duplicates are added.

For this I prefer to add a std::map<line, std::set<>> in the class GtfsData. It's clearer that way, what is only need during data import and what is part on the object model.